WPMedia/fusion-news-theme-blocks
# TMEDIA 793: Update right rail layouts

## Description

Update right rail and advanced right rail layout using the new architecture 

## Jira Ticket

- [TMEDIA-793](https://arcpublishing.atlassian.net/browse/TMEDIA-793)

## Acceptance Criteria

- Update the right-rail and right-rail-advanced layouts so they can be used with blocks on the new architecture 

## Test Steps

1. Checkout this branch `git checkout TMEDIA-793-right-rail-layouts`
2. `npm run storybook`
3. Go to right rail page 
4. Compare with mocks figma https://www.figma.com/file/wQd5edf6Nt1MB3b6Qlzp6x/Arc-Blocks-Design-System?node-id=1007%3A1639
5. Check on mobile and desktop using the news theme
6. Go to advanced right rail 
7. Compare the order with the published version https://5f91e4721ccaba0022a10782-oztmgknobz.chromatic.com/?path=/story/layouts-right-rail-advanced--basic
8. Spacing should match our existing standards in right rail. The mocks spacing in figma trumps our previously-published versions. 

## Effect Of Changes

### Before

- More nesting. Unused classes

### After

- Only styles that are used at a breakpoint are included 

## Dependencies or Side Effects

Feature pack pr: https://github.com/WPMedia/arc-themes-feature-pack/pull/235

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



----
# **Questions:** 

- I'm using the stack in a couple places to allow for customization and gap. However, I did not use the Stack in the instance where I needed to change flex direction. Should that be a behavior of the Stack? I was not able to override the styles via the theme file; based on a breakpoint, we cannot change the props going in to the component.  -> I was using the wrong theme component; I should have been using grid
- Is this method of implementing the layout what was expected for the architecture? Is it ok not to use a theme component?  -> we want to match the mocks rather than the actual 
- Do you have any suggestions for the naming of the interior nested divs of the right rail and the interior main item? -> updated names of fixed width and others where it was matching a pagebuilder name
- I included some `initial` in the values as the breakpoints were not allowing overriding the values. I seem to remember this being fixed. I updated the arc themes components package. I was not seeing the breakpoints updating as expected. Is there an issue with my setup? -> this is only fixed for components, not blocks

## **Notes:** 


- There was some odd behavior with negative margins on the actual layout. I'm still investigating if the actual mocks have the border flush with the bottom of the main item or 24px below it (like the v1 layout) 
<img width="228" alt="Screen Shot 2022-05-09 at 17 08 49" src="https://user-images.githubusercontent.com/5950956/167506786-5848a997-9e3a-4f7e-9f91-0547ad7b1f90.png"> -> border should be flush to the end of the content rather than with the additional 24px
- I have not started work on the advanced layout
- I do not have a feature pack theme token update yet
- I have not updated the tests for this pr -> tests are updated
